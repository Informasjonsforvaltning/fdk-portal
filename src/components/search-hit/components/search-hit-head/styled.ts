import styled from 'styled-components';
import DatasetIconBase from '../../../../images/icon-catalog-dataset-lg.svg';
import ApiIconBase from '../../../../images/icon-catalog-api-lg.svg';
import ConceptIconBase from '../../../../images/icon-catalog-concept-lg.svg';
import InfomodIconBase from '../../../../images/icon-catalog-infomod-lg.svg';
import AuthoritativeIconBase from '../../../../images/icon-authoritative-md.svg';

const Head = styled.div`
    background-color: ${({ theme }) => theme.light};
    border-radius: 5px;
    color: ${({ theme }) => theme.dark};
    display: flex;
    padding: .5em;
    margin-bottom: 1em;
    
    a {
      color: ${({ theme }) => theme.dark} !important;
    }
    svg > path {
      fill: ${({ theme }) => theme.dark};
    }
  }
`;

const HeadTypeIndicator = styled.div`
  display: flex;
  align-items: center;
`;

const HeadInformation = styled.div`
  margin-left: 1em;
`;

const Header = styled.div`
  display: flex;
`;

const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: 600;
  margin-bottom: 0;
`;

const Type = styled.div`
  font-size: 1.6rem;
`;

const DatasetIcon = styled(DatasetIconBase)`
  height: 3em;
`;

const ApiIcon = styled(ApiIconBase)`
  height: 3em;
`;

const ConceptIcon = styled(ConceptIconBase)`
  height: 3em;
`;

const InfomodIcon = styled(InfomodIconBase)`
  height: 3em;
`;

const AuthoritativeIcon = styled(AuthoritativeIconBase)`
  margin-left: 0.5em;
  width: 1.2em;
`;

export default {
  Head,
  HeadTypeIndicator,
  HeadInformation,
  Header,
  Title,
  Type,
  DatasetIcon,
  ApiIcon,
  ConceptIcon,
  InfomodIcon,
  AuthoritativeIcon
};
