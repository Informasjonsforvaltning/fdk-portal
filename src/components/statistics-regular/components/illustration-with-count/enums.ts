export enum Variant {
  COLUMN = 'column'
}
