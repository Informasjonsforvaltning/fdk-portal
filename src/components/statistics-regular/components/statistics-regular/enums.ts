export enum Variant {
  LARGE = 'large',
  XLARGESTRONG = 'xlargeStrong'
}
