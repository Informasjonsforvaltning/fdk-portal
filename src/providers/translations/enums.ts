export enum Cookie {
  LANGUAGE = 'lang'
}
