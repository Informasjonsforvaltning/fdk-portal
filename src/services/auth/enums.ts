export enum OidcError {
  LOGIN_REQUIRED = 'login_required'
}
