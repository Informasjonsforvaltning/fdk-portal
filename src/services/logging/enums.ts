export enum Severity {
  ERROR = 'ERROR',
  WARN = 'WARN',
  INFO = 'INFO'
}
