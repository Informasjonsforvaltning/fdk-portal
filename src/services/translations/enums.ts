export enum Language {
  NB = 'nb',
  NN = 'nn',
  EN = 'en'
}
