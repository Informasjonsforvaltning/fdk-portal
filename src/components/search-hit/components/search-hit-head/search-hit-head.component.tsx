import React, { FC } from 'react';
import { Link } from 'react-router-dom';

import SC from './styled';
import type { TextLanguage } from '../../../../types';
import { SearchTypes } from '../../../../types/enums';
import { getTranslateText } from '../../../../lib/translateText';
import localization from '../../../../lib/localization';
import {
  PATHNAME_CONCEPTS,
  PATHNAME_DATA_SERVICES,
  PATHNAME_DATASETS,
  PATHNAME_INFORMATIONMODELS,
  PATHNAME_PUBLIC_SERVICES,
  PATHNAME_EVENTS
} from '../../../../constants/constants';

interface Props {
  id?: string;
  type: SearchTypes;
  title?: Partial<TextLanguage>;
  subtitle?: React.ReactNode;
  isAuthoritative?: boolean;
}

const detailLinks = {
  [SearchTypes.dataset]: PATHNAME_DATASETS,
  [SearchTypes.dataservice]: PATHNAME_DATA_SERVICES,
  [SearchTypes.concept]: PATHNAME_CONCEPTS,
  [SearchTypes.informationModel]: PATHNAME_INFORMATIONMODELS,
  [SearchTypes.publicService]: PATHNAME_PUBLIC_SERVICES,
  [SearchTypes.event]: PATHNAME_EVENTS
};

export const SearchHitHead: FC<Props> = ({
  id,
  type,
  title,
  subtitle,
  isAuthoritative = false
}) => (
  <SC.Head inverted={type === SearchTypes.event}>
    <SC.HeadTypeIndicator>
      {type === SearchTypes.dataset && <SC.DatasetIcon />}
      {type === SearchTypes.dataservice && <SC.ApiIcon />}
      {type === SearchTypes.concept && <SC.ConceptIcon />}
      {type === SearchTypes.informationModel && <SC.InfomodIcon />}
      {type === SearchTypes.publicService && <SC.ServiceIcon />}
      {type === SearchTypes.event && <SC.ServiceIcon />}
    </SC.HeadTypeIndicator>
    <SC.HeadInformation>
      <SC.Header>
        {title && (
          <SC.Title>
            <Link to={`${detailLinks[type]}/${id}`}>
              {getTranslateText(title)}
            </Link>
          </SC.Title>
        )}
        {isAuthoritative && (
          <div title={localization.authoritativeDatasetTooltip}>
            <SC.AuthoritativeIcon />
          </div>
        )}
      </SC.Header>
      {subtitle && <SC.Type>{subtitle}</SC.Type>}
    </SC.HeadInformation>
  </SC.Head>
);
