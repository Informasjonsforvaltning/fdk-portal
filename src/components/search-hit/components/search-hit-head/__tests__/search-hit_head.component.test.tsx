import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import { SearchHitHead } from '../search-hit-head.component';
import { SearchTypes } from '../../../../../types/enums';

afterEach(cleanup);

describe('SearchHitHead component', () => {
  it('should render with properties', () => {
    const container = shallow(
      <SearchHitHead
        id='123'
        type={SearchTypes.dataset}
        title={{ nb: 'Test datasett' }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
