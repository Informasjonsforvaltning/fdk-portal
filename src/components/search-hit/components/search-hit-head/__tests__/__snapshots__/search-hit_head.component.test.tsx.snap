// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHitHead component should render with properties 1`] = `
<styled.div
  inverted={false}
>
  <styled.div>
    <Styled(SvgIconCatalogDatasetLg) />
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.h2>
        <Link
          to="/datasets/123"
        >
          Test datasett
        </Link>
      </styled.h2>
    </styled.div>
  </styled.div>
</styled.div>
`;
